File: KMeans/src/main/java/com/example/demo1/Classes/Centroid.java
package com.example.demo1.Classes;

import javafx.scene.paint.Color;

import java.util.ArrayList;


public class Centroid extends Dot {

    private Color color;
    private ArrayList<Point> pointArrayList;

    public Centroid(int x, int y, Color color) {
        super(x, y);
        this.color = color;
        pointArrayList = new ArrayList<>();
    }

    public Color getColor() {
        return color;
    }

    public void setColor(Color color) {
        this.color = color;
    }

    public ArrayList<Point> getPointArrayList() {
        return pointArrayList;
    }

    public void setPointArrayList(ArrayList<Point> pointArrayList) {
        this.pointArrayList = pointArrayList;
    }
}

File: Kmeans2/Classes/Centroid.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kmeans2.Classes
{
    public class Centroid : Dot
    {

        private Color color;
        private List<MyPoint> pointArrayList;

        public Centroid(int x, int y, Color color) : base(x, y)
        {
            this.color = color;
            pointArrayList = new List<MyPoint>();
        }

        public Color getColor()
        {
            return color;
        }

        public void setColor(Color color)
        {
            this.color = color;
        }

        public List<MyPoint> getPointArrayList()
        {
            return pointArrayList;
        }

        public void setPointArrayList(List<MyPoint> pointArrayList)
        {
            this.pointArrayList = pointArrayList;
        }
    }
}
